src/pybind/mgr/dashboard_v2/README.rst

```diff
@@ -244,3 +244,52 @@ The settings management implementation will make sure that if you change a
 setting value from the python code you will see that change when accessing
 that setting from the CLI and vice-versa.
 
+
+How to listen for manager notifications in a controller?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The manager notifies the modules with several types of cluster events, such
+as cluster logging event, etc...
+Each module has a "global" handler function called ``notify`` that the manager
+calls to notify the module. But this handler function must not block neither
+spend to much time processing the event notification.
+For this reason we provide a notification queue that controllers can register
+themselves with to receive the cluster notifications.
+
+The example below represents a controller that implements a very simple live
+log viewer page::
+
+  from __future__ import absolute_import
+
+  import collections
+
+  import cherrypy
+
+  from ..tools import ApiController, BaseController, NotificationQueue
+
+
+  @ApiController('livelog')
+  class LiveLog(BaseController):
+      log_buffer = collections.deque(maxlen=1000)
+
+      def __init__(self):
+          super(LiveLog, self).__init__()
+          NotificationQueue.register(self.log, 'clog')
+
+      def log(self, log_struct):
+          self.log_buffer.appendleft(log_struct)
+
+      @cherrypy.expose
+      def default(self):
+          ret = '<html><meta http-equiv="refresh" content="2" /><body>'
+          for l in self.log_buffer:
+              ret += "{}<br>".format(l)
+          ret += "</body></html>"
+          return ret
+
+As you can see above, the ``NotificationQueue`` class provides a register
+method that receives the a function as its first argument, and receives the
+"notification type" as the second argument.
```

Comment:
What notification types are available. Maybe adding a link to the related Ceph docu might be useful.

Reply:
That's a good question! I don't even know where to look. Do you know if already exists a documentation page with that information?

Reply:
I just investigated a bit, and there is no documentation, so I went a bit further and looked into the ceph-mgr C++ code to see if there was a centralized place where these strings are declared, and there isn't. It's all scattered across the code.

What I will do is to collect the several strings that I found in the C++ code and add that list to our README with a disclaimer that those string might change in the future. What do you think?

Reply:
This is better than nothing. But if there is no official docu for that it is not our work to provide it.